src/and-cli-health-check.ts

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env node
+
+import { CommandDefinitions } from "./modules/command-definitions";
+import { CommandRunner } from "./modules/command-runner";
+import { Echo } from "./modules/echo";
+import { Js } from "./modules/js";
+import program from "commander";
+import fetch from "node-fetch";
+import shell from "shelljs";
+import { StringUtils } from "andculturecode-javascript-core";
+
+CommandRunner.run(async () => {
+    // -----------------------------------------------------------------------------------------
+    // #region Constants
+    // -----------------------------------------------------------------------------------------
+
+    const ERROR_MISSING_ENDPOINT = "Please provide a url to verify";
+    const CODE_DESCRIPTION =
+        "Desired response code to wait for. Defaults to 200";
+    const TIMEOUT_DESCRIPTION =
+        "Optional length in ms to wait for the operation to complete";
+    const INTERVAL_DESCRIPTION =
+        "Optional length in ms to wait between operations";
+
+    // #endregion Constants
+    // -----------------------------------------------------------------------------------------
+    // #region Variables
+    // -----------------------------------------------------------------------------------------
+
+    let code: number;
+    let endpoint: string;
+    let responseCode: number;
+    let timeout: number;
+    let interval: number;
+
+    // #endregion Variables
+
+    // -----------------------------------------------------------------------------------------
+    // #region Functions
+    // -----------------------------------------------------------------------------------------
+    const healthCheck = {
+        async checkHost(elapsed: number) {
+            let result = false;
+            Echo.message(
+                `Pinging for HTTP ${code} (${elapsed / 1000}s) of ${timeout /
+                    1000}s...`
+            );
+
+            const response = await fetch(endpoint);
+            responseCode = response.status;
+
+            if (responseCode == code) {
```

Comment:
Thought: Instead of pulling out a `result` boolean, setting it and then checking `result === true`, why not just check `responseCode`? Could reduce simplify some of the logic here... (doesn't look like `responseCode` is used out of this scope, either, so no need to define it above..)

```ts
async checkHost(elapsed: number) {
    Echo.message(
        `Pinging for HTTP ${code} (${elapsed / 1000}s) of ${timeout /
            1000}s...`
    );

    const response = await fetch(endpoint);
    const { status } = response;
    if (status === code) {
        Echo.success("Verified Host!");
        shell.exit(0);
    }

    Echo.message(`Returned with HTTP ${status}. Retrying...`);
},
```